Classes/common/touchdb/TD_Database.m

```diff
@@ -76,18 +79,46 @@ static BOOL removeItemIfExists(NSString* path, NSError** outError)
 
 - (NSString*)attachmentStorePath
 {
-    return [[_path stringByDeletingPathExtension] stringByAppendingString:@" attachments"];
+    return [TD_Database attachmentStorePathWithDatabasePath:_path];
 }
 
-+ (TD_Database*)createEmptyDBAtPath:(NSString*)path
++ (NSString *)attachmentStorePathWithDatabasePath:(NSString *)path
+{
+    return [[path stringByDeletingPathExtension] stringByAppendingString:@" attachments"];
+}
+
++ (instancetype)createEmptyDBAtPath:(NSString*)path
+          withEncryptionKeyProvider:(id<CDTEncryptionKeyProvider>)provider
 {
     if (!removeItemIfExists(path, NULL)) return nil;
     TD_Database* db = [[self alloc] initWithPath:path];
     if (!removeItemIfExists(db.attachmentStorePath, NULL)) return nil;
-    if (![db open]) return nil;
+    if (![db openWithEncryptionKeyProvider:provider]) return nil;
     return db;
 }
 
++ (BOOL)isDatabaseUnencryptedAtPath:(NSString *)path
+{
+    // Load file
+    NSFileHandle *fileHandle = [NSFileHandle fileHandleForReadingAtPath:path];
+    if (!fileHandle) {
+        return YES;
+    }
```

Comment:
At the moment I wrote that `if`, it looked like the right thing; if the db does not exist, then it is not encrypted but likewise we can state the contrary, both options are right. However returning `YES` is the right answer for the code in this [line](https://github.com/cloudant/CDTDatastore/pull/112/files#diff-8e8b4e1d26db1376fbf88a600a1a7a1fR242).

Probably, it would be a better idea to return an `enum` value:
- isUnencrypted
- isEncrypted
- fileNotFound

What do you think?